### ISSUE TITLE
Can this be configure to join as peer?

### ISSUE THREAD

**Issue by cedricve** [klutchell/balena-wireguard#78]
Hey there, is there a possibility to configure this as a peer instead of the Wireguard server?

**Comment by klutchell** [comment]
Unfortunately not with this [entrypoint](https://github.com/klutchell/balena-wireguard/blob/main/entrypoint.sh) script.

The dependencies in the Dockerfile would be unchanged but you would need a different set of configuration and startup steps (probably much simpler) that this project does not currently support.

If you find the required steps to join as a peer and want to open a PR I would be happy to look at it with you!

**Comment by cedricve** [comment]
Yes would love to do it! Let me come back with a plan/suggestion. Thanks @klutchell !

**Comment by cedricve** [comment]
PS: why didn't you upload it to your Balena hub? :P

**Comment by klutchell** [comment]
> PS: why didn't you upload it to your Balena hub? :P

Good question! I think because I'm not actively maintaining it at the moment so I didn't want an influx of users looking for support and features?

That said, I would be happy to support you as a contributor and clean up the project in the process. I just happen to use [tailscale](https://github.com/klutchell/balena-tailscale) as my primary VPN now and don't have use for manual wireguard configuration.

**Comment by cedricve** [comment]
that's cool! Does this one acts as a subnet gateway? As what I'm proposing in this issue

**Comment by klutchell** [comment]
yes, the tailscale client can act as a subnet gateway with the correct flags applied

**Comment by cedricve** [comment]
I'm currently using Netmaker which is self-hosted, I just read about headscale. Will give it some thoughts! ;) Thanks for the discussion
